File: src/Services/Customers/FoodDelivery.Services.Customers/Products/Models/Product.cs
using FoodDelivery.Services.Customers.Shared.Clients.Rest.Catalogs.Dtos;

namespace FoodDelivery.Services.Customers.Products.Models;

public record Product
{
    public long Id { get; init; }
    public string Name { get; init; } = default!;
    public string? Description { get; init; }
    public decimal Price { get; init; }
    public long CategoryId { get; init; }
    public string CategoryName { get; init; } = default!;
    public long SupplierId { get; init; }
    public string SupplierName { get; init; } = default!;
    public long BrandId { get; init; }
    public string BrandName { get; init; } = default!;
    public int AvailableStock { get; init; }
    public int RestockThreshold { get; init; }
    public int MaxStockThreshold { get; init; }
    public ProductStatus ProductStatus { get; init; }
    public int Height { get; init; }
    public int Width { get; init; }
    public int Depth { get; init; }
}

File: src/UIs/Spa/react-food-delivery/src/features/products/models/product.ts
export interface ProductImage {
  id: number;
  productId: number;
  imageUrl: string;
  isMain: boolean;
}

export interface Product {
  id: number;
  name: string;
  price: number;
  stock: number;
  restockThreshold: number;
  maxStockThreshold: number;
  status: number;
  productType: number;
  productColor: number;
  height: number;
  width: number;
  depth: number;
  size: string;
  categoryId: number;
  categoryName: string;
  supplierId: number;
  supplierName: string;
  brandId: number;
  brandName: string;
  description?: string;
  images?: ProductImage[];
}